opencog/atoms/base/Atom.h

```diff
@@ -190,6 +195,16 @@ class Atom
         return classserver().isA(at, t);
     }
 
+    /** Returns the outgoing set.
+     *
+     * @return The outgoing set.
+     */
+    virtual inline const HandleSeq& getOutgoingSet() const
+    {
+        static HandleSeq no_atoms;
+        return no_atoms;
+    }
```

Comment:
The unit tests would pass since the code could never have been executed before because the call wasn't available to Atom only through NodeCast of handles which were checked for null.

If we do a throw here, I'll have to catch them in Python and then do something else anyway since name is a **property** of Atom and needs to be accessible for all atoms. Forcing people to do is instance checks before using a property of an object would be very odd in Python.